File: bankingOnGolang/internal/model/RepaymentSchedule.go
package model

import (
	"time"
)

// ==============================================================
// RepaymentSchedule Declaration
// ==============================================================
type RepaymentSchedule struct {
	BaseModel
	InstallmentNumber int32
	DueDate           time.Time
	PrincipalDue      Money `gorm:"embedded;embeddedPrefix:repaymentSchedule_principalDue"`
	InterestDue       Money `gorm:"embedded;embeddedPrefix:repaymentSchedule_interestDue"`
	TotalDue          Money `gorm:"embedded;embeddedPrefix:repaymentSchedule_totalDue"`
	LoanAccountId     *uint
	LoanAccount       *LoanAccount `gorm:"foreignKey:LoanAccountId"`
	PaymentId         *uint
	Payment           *LoanPayment `gorm:"foreignKey:PaymentId"`
	Status            InstallmentStatus

	// parent associations as their child

}

File: src/app/models/RepaymentSchedule.ts


// Define collection and schema for RepaymentSchedule
export interface RepaymentSchedule {
    installmentNumber: number
    dueDate: Date
    principalDue: Money
    interestDue: Money
    totalDue: Money
    LoanAccount: Schema.Types.ObjectId
    Payment: Schema.Types.ObjectId
    Status:  String
    collection: 'repaymentSchedules'
}
